pkg/dns/azure/dns_test.go

```diff
@@ -96,3 +99,55 @@ func TestDeleteDNS(t *testing.T) {
 		t.Fatalf("expected the dns client 'Delete' func to be called, but found %s instead", recordedCall)
 	}
 }
+
+func TestGetTagList(t *testing.T) {
+	infra := configv1.Infrastructure{
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName: "test",
+			PlatformStatus: &configv1.PlatformStatus{
+				Azure: &configv1.AzurePlatformStatus{
+					ResourceGroupName: "test-rg",
+					CloudName:         configv1.AzureStackCloud,
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name         string
```

Comment:
If this should be doing something slightly different for public vs private zones, please make some tests with zones included.

Reply:
Tagset is common for both zone types.